python/tank/deploy/tank_commands/update.py

```diff
@@ -149,80 +217,92 @@ def run_interactive(self, log, args):
-            else:
-                # the item we are filtering on does not exist in this env
-                engines_to_process = []
-        
-        for engine in engines_to_process:
-            items.append( _process_item(log, suppress_prompts, tk, env, engine) )
-            log.info("")
+        # now process them one after the other
+        for env_filename in env_filenames: 
+            env_obj = WritableEnvironment(env_filename, pc)
+            env_obj.use_preserving_yaml_parser(preserve_yaml)
             
-            if app_instance_name is None:
-                # no filter - process all apps
-                apps_to_process = env.get_apps(engine)
-            else:
-                # there is a filter! Ensure the filter matches
-                # something in the current engine apps listing
-                if app_instance_name in env.get_apps(engine):
-                    # the filter matches something!
-                    apps_to_process = [app_instance_name]
-                else:
-                    # the app filter does not match anything in this engine
-                    apps_to_process = []
+            updated_items += _process_environment(tk, 
+                                                  log, 
+                                                  env_obj, 
+                                                  engine_instance_name, 
+                                                  app_instance_name, 
+                                                  suppress_prompts)
             
-            for app in apps_to_process:
-                items.append( _process_item(log, suppress_prompts, tk, env, engine, app) )
-                log.info("")
-        
-        if len(env.get_frameworks()) > 0:
-            log.info("")
-            log.info("Frameworks:")
-            log.info("-" * 70)
-
-            for framework in env.get_frameworks():
-                items.append( _process_item(log, suppress_prompts, tk, env, framework_name=framework) )
-        
+    else:
 
+        # process non-external config 
+        if env_name is None:
+            env_names_to_process = pc.get_environments()
+        else:
+            env_names_to_process = [env_name]
+    
+        for env_name in env_names_to_process:
+            env_obj = pc.get_environment(env_name, writable=True)
+            env_obj.use_preserving_yaml_parser(preserve_yaml)
+            
+            updated_items += _process_environment(tk, 
+                                                  log, 
+                                                  env_obj, 
+                                                  engine_instance_name, 
+                                                  app_instance_name, 
+                                                  suppress_prompts)
+    
+    
     # display summary
     log.info("")
     summary = []
-    for x in items:
+    for x in updated_items:
         if x["was_updated"]:
```

Comment:
i think this was wrong - i have changed `updated_items` to `processed_items` for clarity.